### ISSUE TITLE
More privacy on protected contents...

### ISSUE THREAD

**Issue by 411A** [telegramdesktop/tdesktop#17411]
### Is your feature request related to a problem?

There are some features that compromise users privacy.
1️⃣ We shouldn't be able to select the text from the protected
channel (though `Ctrl + C` is disabled, the ability to select the text is nonsense).
2️⃣ We shouldn't be able to take screenshots from protected
content (it also allow us to do OCR & extract the texts).
3️⃣ Files downloaded from a protected channel should be cached
in a hidden directory or a directory that is hard to access for regular users (e.g. **WindowsApps** folder) & should be deleted if the user didn't access them at a certain time (the music hadn't played from Telegram for more than a week).
![image](https://user-images.githubusercontent.com/13054619/146547712-22488ddf-be6c-4b52-b6d0-076c39c39d66.png)

### Describe the solution you'd like

The most important thing to consider on the next update is to **disable the screenshot from protected channels**. This is already done in [Unigram](https://www.microsoft.com/store/productId/9N97ZCKPD60Q) app by enabling the `WDA_EXCLUDEFROMCAPTURE` flag from [SetWindowDisplayAffinity](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowdisplayaffinity) function.

### Describe alternatives you've considered

Enabling the `WDA_EXCLUDEFROMCAPTURE` flag from [SetWindowDisplayAffinity](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowdisplayaffinity) function, & other necessary functions ([DwmIsCompositionEnabled](https://docs.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmiscompositionenabled)).

### Additional context

⭐ Ability to start a secret chat & use it from both mobile &
desktop (disabled screenshot & caching).

**Comment by john-preston** [comment]
1. I don't agree. Non-selectable text looks like a UI bug, while this way you can show a clear message about what's going on.
3. Media (photos and videos) are used from media cache only, while files are available - no sense in keeping them somewhere else, you can always "Show In Folder" them or open in an external app and "Save As..." somewhere.

2. I need to test that, thanks for the API reference.

**Comment by nuclight** [comment]
0. Don't use this shit emoji for numbering.
1. Many people use selecting for marking place where they were reading long messages, e.g. after switching window  to another app.
2. Your suggested API is Windows-only (and Unigram is Windows-only), but Telegram Desktop is cross-platform - non-OS-specific feature must work either on all supported operating systems or none at all.
3. Again, OS considerations from p.2 apply, but also with other Telegram clients - if e.g. Android version allows access to cache, then why Desktop should be different? You must understand that absolute protection is impossible and is always breakable if user have enough "skills level" (e.g. most users don't know how to go to cache). If authors of the platform in their main product considered some level enough (mobile is first-class for them, desktop is second), then why Desktop developer should spend effort to higher - yet easily breakable - levels, instead of more important things?

**Comment by GregorKopka** [comment]
Given that one can always take a mobile to take picture or video of a screen... no 100% protection is possible.
The main use I personally see is that one can't forward messages from such chats, thus it is not possible to create a 'digital paper trail' like with normal forwarding (where the account that created the content is linked to the forwarded message, leading to anyone having it as a contact can verify it it really is the origin).

**Comment by nuclight** [comment]
> thus it is not possible to create a 'digital paper trail' like with normal forwarding (where the account that created the content is linked to the forwarded message, leading to anyone having it as a contact can verify it it really is the origin).

Given the world where fakes constitute the more and more serious problem, having 'digital paper trail' and verification is actually a Good Thing to fight with evil - there must be no such thing as "forbid to verified copy" in ideal new world.

**Comment by GregorKopka** [comment]
> Given the world where fakes constitute the more and more serious problem, having 'digital paper trail' and verification is actually a Good Thing to fight with evil - there must be no such thing as "forbid to verified copy" in ideal new world.

That _new world_ has been described in a book written by George Orwell, which was a warning and not a manual.
Please stop confusing that.

Topic is about privacy, not about nuking it.

**Comment by Zx-EvM** [comment]
Is there any news about it? For now the feature that works only for mobile apps looks pretty pointless, isn't?

**Comment by BearusuMyo** [comment]
A way to solve this could be just treating those chats like secret chats are and make 'em viewable at mobile platforms only.

**Comment by nuclight** [comment]
> > Given the world where fakes constitute the more and more serious problem, having 'digital paper trail' and verification is actually a Good Thing to fight with evil - there must be no such thing as "forbid to verified copy" in ideal new world.
> 
> That _new world_ has been described in a book written by George Orwell, which was a warning and not a manual. Please stop confusing that.
> 
> Topic is about privacy, not about nuking it.

No, that's you who are confusing ideal new world with Orwell's "new world" (BTW, that term was not from Orwell but Huxsely). However, your so called "privacy" here proceeds us to both Orwell/Huxsley, not to what *should* be.
